src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Assignment.cs

```diff
@@ -14,10 +16,12 @@ public ExprAssignment(Expr lhs, Expr rhs)
             Flags = EXPRFLAG.EXF_ASSGOP;
         }
 
-        public Expr LHS { get; set; }
+        public Expr LHS
+        {
+            get => _lhs;
+            set => Type = (_lhs = value).Type;
```

Comment:
I wonder if we set LHS anywhere outside construction? 
Would be nice to just set this in the ctor and make readonly.

---
In reply to: [129837030](https://github.com/dotnet/corefx/pull/22684#discussion_r129837030) [](ancestors = 129837030)